homeassistant/components/zwave_mqtt/entity.py

```diff
@@ -0,0 +1,272 @@
+"""Generic Z-Wave Entity Classes."""
+
+import copy
+import logging
+
+from openzwavemqtt.const import EVENT_INSTANCE_STATUS_CHANGED, EVENT_VALUE_CHANGED
+from openzwavemqtt.models.node import OZWNode
+
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import Entity
+
+from . import const
+from .const import DOMAIN, PLATFORMS
+from .discovery import check_node_schema, check_value_schema
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZWaveDeviceEntityValues:
+    """Manages entity access to the underlying Z-Wave value objects."""
+
+    def __init__(self, hass, options, schema, primary_value):
+        """Initialize the values object with the passed entity schema."""
+        self._hass = hass
+        self._entity_created = False
+        self._schema = copy.deepcopy(schema)
+        self._values = {}
+        self.options = options
+
+        # Go through values listed in the discovery schema, initialize them,
+        # and add a check to the schema to make sure the Instance matches.
+        for name, disc_settings in self._schema[const.DISC_VALUES].items():
+            self._values[name] = None
+            disc_settings[const.DISC_INSTANCE] = [primary_value.instance]
+
+        self._values[const.DISC_PRIMARY] = primary_value
+        self._node = primary_value.node
+        self._schema[const.DISC_NODE_ID] = [self._node.node_id]
+
+    def setup(self):
+        """Set up values instance."""
+        # Check values that have already been discovered for node
+        # and see if they match the schema and need added to the entity.
+        for value in self._node.values():
+            self.check_value(value)
+
+        # Check if all the _required_ values in the schema are present and
+        # create the entity.
+        self._check_entity_ready()
+
+    def __getattr__(self, name):
```

Comment:
are value names static?

Reply:
no, they can be changed in OZW config files. We can make them static by not looking at the label but at the id. We have enums with all current command classes and values.

Reply:
if they can be changed in OZW, we shouldn't use them as attributes I think?

Reply:
Values are different for each CommandClass this will auto generate the attributes for each commandclass value.

Btw: Almost sure that you wrote this code ;-)

Reply:
Ignore my previous comment about the attributes being changes OZW-side. Those value are not being translated by us and we use the ID's (list values)

Reply:
We'll keep this for now.